src/core/mac/mac.cpp

```diff
@@ -856,10 +856,116 @@ void Mac::HandleBeginTransmit(void)
 
     if (error != kThreadError_None)
     {
+#if OPENTHREAD_ENABLE_PASSING_UP_ACK
+        TransmitDoneTask(mTxFrame, NULL, kThreadError_Abort);
+#else
         TransmitDoneTask(mTxFrame, false, kThreadError_Abort);
+#endif
+    }
+}
+
+#if OPENTHREAD_ENABLE_PASSING_UP_ACK
```

Comment:
What is the reasoning behind making this configurable? Could we not make it a requirement?

Also, there is a lot of code split by this \#if \#else.  That will create more work going forward for anyone that needs to change this code as they will have to make the same changes in two places.  Could we instead refactor this so that the portion of code that is within the \#if \#else is as small as possible?  A couple ways I can see to do that are; 1) use multiple \#if's, 2) create a new function that contains the common code and is called by both versions.

I prefer option 2 myself.